cns/service/main.go

```diff
@@ -1182,34 +1184,41 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	}
 	logger.Printf("reconciled initial CNS state after %d attempts", attempt)
 
-	// the nodeScopedCache sets Selector options on the Manager cache which are used
+	scheme := kuberuntime.NewScheme()
+	if err := corev1.AddToScheme(scheme); err != nil { //nolint:govet // intentional shadow
+		return errors.Wrap(err, "failed to add corev1 to scheme")
+	}
+	if err = v1alpha.AddToScheme(scheme); err != nil {
+		return errors.Wrap(err, "failed to add nodenetworkconfig/v1alpha to scheme")
+	}
+	if err = v1alpha1.AddToScheme(scheme); err != nil {
+		return errors.Wrap(err, "failed to add clustersubnetstate/v1alpha1 to scheme")
+	}
+
+	// Set Selector options on the Manager cache which are used
 	// to perform *server-side* filtering of the cached objects. This is very important
 	// for high node/pod count clusters, as it keeps us from watching objects at the
 	// whole cluster scope when we are only interested in the Node's scope.
-	nodeScopedCache := cache.BuilderWithOptions(cache.Options{
-		SelectorsByObject: cache.SelectorsByObject{
+	cacheOpts := cache.Options{
+		Scheme: scheme,
+		ByObject: map[client.Object]cache.ByObject{
 			&v1alpha.NodeNetworkConfig{}: {
-				Field: fields.SelectorFromSet(fields.Set{"metadata.name": nodeName}),
+				Field: fields.SelectorFromSet(fields.Set{"metadata.name": nodeName, "metadata.namespace": "kube-system"}),
 			},
 			&corev1.Pod{}: {
 				Field: fields.SelectorFromSet(fields.Set{"spec.nodeName": nodeName}),
 			},
 		},
-	})
-
-	crdSchemes := kuberuntime.NewScheme()
-	if err = v1alpha.AddToScheme(crdSchemes); err != nil {
-		return errors.Wrap(err, "failed to add nodenetworkconfig/v1alpha to scheme")
-	}
-	if err = v1alpha1.AddToScheme(crdSchemes); err != nil {
-		return errors.Wrap(err, "failed to add clustersubnetstate/v1alpha1 to scheme")
 	}
-	manager, err := ctrl.NewManager(kubeConfig, ctrl.Options{
-		Scheme:             crdSchemes,
+
+	managerOpts := ctrlmgr.Options{
+		Scheme:             scheme,
 		MetricsBindAddress: "0",
-		Namespace:          "kube-system", // TODO(rbtr): namespace should be in the cns config
-		NewCache:           nodeScopedCache,
-	})
+		Cache:              cacheOpts,
+		Logger:             ctrlzap.New(),
```

Comment:
where are we going to need to use this manager's logger?

Reply:
it is used internally by the manager to log things like reconcile events queued